pkg/sql/drop_sequence.go

```diff
@@ -227,16 +227,54 @@ func (p *planner) canRemoveOwnedSequencesImpl(
 // given sequence descriptor being dropped, and if the dependent object
 // is a view, it drops the views.
 // This is called when the DropBehavior is DropCascade.
+//
+// Note: It is perfectly fine to have a sequence depended on by a constraint in another
+// table, in which case we *should* drop that depending constraint in the depending
+// table as well, but we do not do it here (we skip those depending tables). It's a limitation
+// and should be addressed, once we decided to come back to address the bigger issue of
+// sequence reference/management in the legacy schema changer.
+// The `numDependedOnByTablesToSkip` variable is used exactly to achieve this temporary hack.
 func dropDependentOnSequence(ctx context.Context, p *planner, seqDesc *tabledesc.Mutable) error {
-	for len(seqDesc.DependedOnBy) > 0 {
-		dependent := seqDesc.DependedOnBy[0]
+	// numDependedOnByTablesToSkip is a temporary hack to skip those tables that reference
```

Comment:
Instead of describing a hack as a temporary, describe the conditions which need to be fulfilled to remove the hack. One should always assume that, unless somebody's actively working on part of the code right now, it's not going to change soon.